core/templates/dev/head/profile/Profile.js

```diff
@@ -31,22 +31,83 @@ oppia.controller('Profile', [
     $rootScope.loadingMessage = 'Loading';
     $http.get(profileDataUrl).success(function(data) {
       $rootScope.loadingMessage = '';
+      $scope.username = {
+        title: 'Username',
+        value: data.username,
+        helpText: (data.username)
+      };
       $scope.userBio = data.user_bio;
       $scope.userDisplayedStatistics = [{
-        title: 'User Impact Score',
+        title: 'Impact',
         value: data.user_impact_score,
         helpText: (
           'A rough measure of the impact of explorations created by this ' +
           'user. Better ratings and more playthroughs improve this score.')
       }, {
-        title: 'Created Explorations',
+        title: 'Created',
         value: data.created_exp_summary_dicts.length
       }, {
-        title: 'Edited Explorations',
+        title: 'Edited',
         value: data.edited_exp_summary_dicts.length
       }];
-      $scope.userCreatedExplorations = data.created_exp_summary_dicts;
-      $scope.userEditedExplorations = data.edited_exp_summary_dicts;
+
+      $scope.userEditedExplorations = data.edited_exp_summary_dicts.sort(
+          function(exploration1, exploration2) {
+        if (exploration1.ratings > exploration2.ratings) {
+          return 1;
+        } else if (exploration1.ratings === exploration2.ratings) {
+          if (exploration1.playthroughs > exploration2.playthroughs) {
+            return 1;
+          } else {
+            return -1;
+          }
+        } else {
+          return -1;
+        }
+      });
+
+      $scope.currentPage = 0;
+      $scope.previousPage = function() {
+        $scope.currentPage--;
+      };
+      $scope.nextPage = function() {
+        $scope.currentPage++;
+      };
+      $scope.pageSize = 6;
+
+      $scope.userDisplayedExplorations = function() {
+        $scope.explorationsOnPage = [];
+        $scope.explorationIndexStart = $scope.currentPage * $scope.pageSize;
+        $scope.explorationIndexEnd = (
+          $scope.explorationIndexStart + $scope.pageSize - 1);
+        for (var count = $scope.explorationIndexStart;
+            count <= $scope.explorationIndexEnd; count++) {
+          var nextExploration = $scope.userEditedExplorations[count];
```

Comment:
It's not so nice to refer to an element that may not be there. Instead, I suggest changing $scope.explorationIndexEnd to be the minimum of what it currently is, and the length of $scope.userEditedExplorations - 1.